click/formatting.py

```diff
@@ -133,14 +123,23 @@ def write_usage(self, prog, args='', prefix='Usage: '):
         :param args: whitespace separated list of arguments.
         :param prefix: the prefix for the first line.
         """
-        prefix = '%*s%s' % (self.current_indent, prefix, prog)
-        self.write(prefix)
-
-        text_width = max(self.width - self.current_indent - term_len(prefix), 10)
-        indent = ' ' * (term_len(prefix) + 1)
-        self.write(wrap_text(args, text_width,
-                             initial_indent=' ',
-                             subsequent_indent=indent))
+        prefix = '%*s%s ' % (self.current_indent, prefix, prog)
+        text_width = self.width - self.current_indent
+
+        if text_width >= (term_len(prefix) + 20):
+            # The arguments will fit to the right of the prefix.
+            indent = ' ' * term_len(prefix)
+            self.write(wrap_text(args, text_width,
+                                 initial_indent=prefix,
+                                 subsequent_indent=indent))
+        else:
+            # The prefix is too long, put the arguments on the next line.
+            self.write(prefix)
+            self.write('\n')
+            indent = ' ' * (self.current_indent + 12)
```

Comment:
Right, I meant the original meaning of prefix, which is just the `"Usage: "` part, not including the program name.

This gives the variable a new meaning:
    prefix = '%*s%s' % (self.current_indent, prefix, prog)